### ISSUE TITLE
nimble install fails to read package info due to bad snap command

### ISSUE THREAD

**Issue by yang573** [sirredbeard/nim_lang_snap#13]
I believe commands that try to evaluate a .nimble file fail due to a bad snap command:

```
> nimble install nimble --debug
    Reading official package list
Downloading https://github.com/nim-lang/nimble using git
    Cloning latest tagged version: v0.12.0
  Executing git clone --recursive --depth 1 -b v0.12.0 https://github.com/nim-lang/nimble /tmp/nimble_64533/githubcom_nimlangnimble
     Output Cloning into '/tmp/nimble_64533/githubcom_nimlangnimble'...
        ... Note: switching to '8f7af860c5ce9634af880a7081c6435e1f2a5148'.
        ...
     Debug: Executing /usr/bin/snap e --hints:off --verbosity:0 --colors:on  /tmp/nimblecache-2141250259/nimble_527315691/nimble.nims /tmp/nimble_64533/githubcom_nimlangnimble/nimble.nimble /tmp/nimble_64533.out printPkgInfo
   Warning: Not removing temporary path because of debug verbosity: /tmp/nimble_64533
     Error: Could not read package info file in /tmp/nimble_64533/githubcom_nimlangnimble/nimble.nimble;
        ...   Reading as ini file failed with: 
        ...     Invalid section: .
        ...   Evaluating as NimScript file failed with: 
        ...     error: unknown command "e", see 'snap help'.
        ... printPkgInfo() failed.
```

It seems that nimble tries to run "> snap e [args]", which fails because it's an unrecognized command.

**Comment by sirredbeard** [comment]
I can reproduce. 

Steps to reproduce:

```
$ sudo snap install nim-lang --classic
$ sudo snap alias nim-lang.nim nim
$ nim-lang.nimble refresh
$ nim-lang.nimble install nimx
```

nimble refresh works, creating a cache of packages of ~/.nimble/,,,

It is downloading the nimble package to /tmp/nimble_xxxx/,,, and something is going wrong there.

I don't call 'snap e' anywhere in the snapcraft.yaml folder, or add an 'e' as an option for nimble.

Here is the error I can reproduce:

```
hayden@hayden-Virtual-Machine:~$ nim-lang.nimble install nimx --verbose --debug
    Reading official package list
Downloading https://github.com/yglukhov/nimx using git
  Executing git clone --recursive --depth 1 https://github.com/yglukhov/nimx /tmp/nimble_37042/githubcom_yglukhovnimx
     Output Cloning into '/tmp/nimble_37042/githubcom_yglukhovnimx'...
     Debug: Executing /usr/bin/snap e --hints:off --verbosity:0 --colors:on  /tmp/nimblecache-1687995341/nimx_3957814991/nimx.nims /tmp/nimble_37042/githubcom_yglukhovnimx/nimx.nimble /tmp/nimble_37042.out printPkgInfo
   Warning: Not removing temporary path because of debug verbosity: /tmp/nimble_37042
     Error: Could not read package info file in /tmp/nimble_37042/githubcom_yglukhovnimx/nimx.nimble;
        ...   Reading as ini file failed with: 
        ...     Invalid section: .
        ...   Evaluating as NimScript file failed with: 
        ...     error: unknown command "e", see 'snap help'.
        ... printPkgInfo() failed.
```

The nimble package files are downloaded in `/tmp/nimble_37042/githubcom_yglukhovnimx/`:

![image](https://user-images.githubusercontent.com/33820650/101977757-ec3beb00-3c1d-11eb-98f1-bfc2c0a93719.png)

Nothing looks like an invalid section in the nimble package `/tmp/nimble_37042/githubcom_yglukhovnimx/nimx.nimble`:

![image](https://user-images.githubusercontent.com/33820650/101977812-52c10900-3c1e-11eb-99e3-9b5f299ed609.png)

I took this strace:

[strace.log](https://github.com/sirredbeard/nim_lang_snap/files/5682608/strace.log)

This looks odd in the strace, it's looking for a .ini that isn't there:

![image](https://user-images.githubusercontent.com/33820650/101977899-fe6a5900-3c1e-11eb-9f1e-05c0256aad8e.png)

Here is the respective code in nimble/packageparser.nim: https://github.com/nim-lang/nimble/blob/27d1063febe3055252103cacb11787ac83893416/src/nimblepkg/packageparser.nim#L384

I think it's trying the ini approach and when that fails, it is trying to run the file as NimScript. That is where the extra 'e' comes from.

But why is it calling /bin/snap instead of nim-lang.nim or our alias nim?

**Comment by sirredbeard** [comment]
The command being run is here https://github.com/nim-lang/nimble/blob/b012df72fd30a2047ef98cd8c60a3b00d813ee01/src/nimblepkg/nimscriptwrapper.nim#L51

getNimBin() is picking up /bin/snap instead of nim. 

getNimBin() appears to be imported from options.nim https://github.com/nim-lang/nimble/blob/ca6fb5bfdf7859061aa71ddad580c87f63f9e902/src/nimblepkg/options.nim#L352

**Comment by sirredbeard** [comment]
Fixed by #15
